src/process.cr

```diff
@@ -415,6 +415,72 @@ class Process
   private def close_io(io)
     io.close if io
   end
+
+  # Sets the real, effective, and saved user to the one specified.
+  def self.become(user : System::User) : Nil
+    return if LibC.setuid(user.uid) == 0
+    Errno.new("The calling process was not privileged.")
```

Comment:
```cr
if LibC.setuid(user.id) != 0
  raise Errno.new("Error setting uid")
end
```

is how this should be used. The descriptive message is for describing what was attempted, not what failed (Errno will format the error string nicely for us). Also you forgot a `raise`. I suggest you add a spec for this using `expect_raises`.

Reply:
Fixed, and spec'ed.